includes/wikia/ExternalUser_Wikia.php

```diff
@@ -215,11 +215,17 @@ public function authenticate( $sPassword ) {
 		$bMediaWikiResult = User::comparePasswords( $this->getPassword(), $sPassword, $this->getId() );
 		// Detect discrepancies between Helios and MediaWiki results.
 		if ( $wgEnableHeliosExt && (  $bHeliosResult != $bMediaWikiResult ) ) {
+			
+			$sMWHashFirst = substr( $this->getPassword(), 0, 3 ); // The first three bytes.
+			$sMWHashLast  = substr( $this->getPassword(),   -3 ); // The last three bytes.
+
 			\Wikia\Helios\User::debugLogin( $sPassword, __METHOD__ );
 			\Wikia\Logger\WikiaLogger::instance()->error(
 				'HELIOS_LOGIN',
 				[ 'helios' => $bHeliosResult, 'mediawiki' => $bMediaWikiResult,
-				 'user_id' => $this->getId(), 'username' => $this->getName() ]
+				 'user_id' => $this->getId(), 'username' => $this->getName(),
+				 'mw_hash_first' => $sMWHashFirst, 'mw_hash_last' => $sMWHashLast,
+				 'mw_hash_length' => strlen( $this->getPassword() ) ]
```

Comment:
I'm getting picky here, but wouldn't it make sense to put each key / value pair in a separate line for readability?